### PR TITLE
WatersAPI transition to React-Query implementation

### PR DESCRIPTION
**Pull request by kevinhaube**
This PR converts usage of the waters api (`/validate` and `/upload`) to use react-query mutations

Test Steps:
1. Ensure `yarn test` runs without failures
2. Load the app locally
3. Ensure the Validate page validates and reports file warnings/errors properly (no functionality differences)

## Changes
- Swap Waters API to React Query implementation

### Testing
- [X] Tested locally?
- [ ] Ran `./prime test` or `./gradlew testSmoke` against local Docker ReportStream container?
- [X] (For Changes to /frontend-react/...) Ran `npm run lint:write`? 
- [X] Added tests?

### Process
- [ ] Are there licensing issues with any new dependencies introduced?
- [ ] Includes a summary of what a code reviewer should test/verify?
- [ ] Updated the release notes?
- [ ] Database changes are submitted as a separate PR?
- [ ] DevOps team has been notified if PR requires ops support?

## Linked Issues
- Fixes #issue

## To Be Done
*Create GitHub issues to track the work remaining, if any*
- #issue

## Specific Security-related subjects a reviewer should pay specific attention to
- Does this PR introduce new endpoints?
    - new endpoint A
    - new endpoint B
- Does this PR include changes in authentication and/or authorization of existing endpoints?
- Does this change introduce new dependencies that need vetting?
- Does this change require changes to our infrastructure?
- Does logging contain sensitive data?
- Does this PR include or remove any sensitive information itself?

If you answered '_yes_' to any of the questions above, conduct a detailed Review that addresses at least:

- What are the potential security threats and mitigations? Please list the _STRIDE_ threats and how they are mitigated
    - **S**poofing (faking authenticity)
        - Threat _T_, which could be achieved by _A_, is mitigated by _M_
    - **T**ampering (influence or sabotage the integrity of information, data, or system)
    - **R**epudiation (the ability to dispute the origin or originator of an action)
    - **I**nformation disclosure (data made available to entities who should not have it)
    - **D**enial of service (make a resource unavailable)
    - **E**levation of Privilege (reduce restrictions that apply or gain privileges one should not have)
- Have you ensured logging does not contain sensitive data?
- Have you received any additional approvals needed for this change?

### CODE REVIEW

File: frontend-react/src/__mocks__/WatersMockServer.ts [hunk 1]
```diff
@@ -70,6 +70,8 @@ const watersResponseError = {
 
 const handlers = [
     rest.post(`${RS_API_URL}/api/waters`, (req, res, ctx) => {
+        if (req.headers["_headers"]["client"] === "test-endpoint-name")
```
Comment (penny-lischer) [inline frontend-react/src/__mocks__/WatersMockServer.ts:73]: Can we add "test-endpoint-name", bad-client", etc... into an enum

Comment (kevinhaube) [reply frontend-react/src/__mocks__/WatersMockServer.ts]: Oo, good call! Updated 👍🏼

**Review by penny-lischer — APPROVED** [review]
LGTM

File: frontend-react/src/hooks/network/WatersHooks.ts [hunk 2]
```diff
@@ -0,0 +1,58 @@
+import { useMutation } from "@tanstack/react-query";
```
Comment (doug-s-nava) [inline frontend-react/src/hooks/network/WatersHooks.ts:1]: fwiw I would maybe have moved this functionality into the `useFileHandler` hook somehow? but this is good too

**Review by doug-s-nava — COMMENTED** [review]
👍

File: frontend-react/src/hooks/network/WatersHooks.ts (hunk 2 above)
Comment (kevinhaube) [reply frontend-react/src/hooks/network/WatersHooks.ts:1]: Along with some rearrangement of stuff post Deliveries react-query work, I want to go back through and assess _where_ some things are called, so I can add this to the list alongside how we handle the Report Details page in the Delivery work.

Comment (kevinhaube) [reply frontend-react/src/hooks/network/WatersHooks.ts:1]: For reference: https://github.com/CDCgov/prime-reportstream/pull/6977#discussion_r993768855

**Pull request merged**